aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2773,29 +2779,27 @@ impl AptosVM {
 /// Transaction execution: AptosVM
 /// Executing conflicts: in the input order, via BlockSTM,
 /// State: BlockSTM-provided MVHashMap-based view with caching
-pub struct AptosVMBlockExecutor;
+pub struct AptosVMBlockExecutor {
+    /// Manages module cache and execution environment of this block executor. Users of executor
+    /// must use manager's API to ensure the correct state of caches.
+    module_cache_manager:
+        ModuleCacheManager<HashValue, ModuleId, CompiledModule, Module, AptosModuleExtension>,
+}
 
-// Executor external API
 impl VMBlockExecutor for AptosVMBlockExecutor {
-    // NOTE: At the moment there are no persistent caches that live past the end of a block (that's
-    // why AptosVMBlockExecutor has no state)
-    // There are some cache invalidation issues around transactions publishing code that need to be
-    // sorted out before that's possible.
-
     fn new() -> Self {
-        Self
+        Self {
+            module_cache_manager: ModuleCacheManager::new(),
+        }
     }
 
-    /// Execute a block of `transactions`. The output vector will have the exact same length as the
-    /// input vector. The discarded transactions will be marked as `TransactionStatus::Discard` and
-    /// have an empty `WriteSet`. Also `state_view` is immutable, and does not have interior
-    /// mutability. Writes to be applied to the data view are encoded in the write set part of a
-    /// transaction output.
```

Comment:
Seems like a very irrelevant place for thus comment. Maybe some parts of it could stay, but it quickly becomes legacy now that we have block limits, retries, block itself can fail. Should I bring it back?

Reply:
Returned the comment, moved it to trait definition, with minor changes.